### ISSUE TITLE
Facets not appearing on keyword search results

### ISSUE THREAD

**Issue by sammyjava** [legumeinfo/mine-issues#71]
Posting this here so I can record the solution.

On 7/27/22 13:51, Andrew Farmer wrote:
> I whacked briefly and seem to have discovered that we've lost facets for organism in keyword search results?
> e.g. https://dev.lis.ncgr.org/cicermine/keywordSearchResults.do?searchTerm=bHLH&searchBag=&facet_Category=Gene
> unless I'm misremembering something about how this should work.

**Comment by sammyjava** [comment]
Aha. Fixed it. So, you'll recall that I enthusiastically did indexing timing tests using haldane so I was working with smaller delays. And I divined that this line was blowing up indexing because every single BioEntity was getting an Organism reference.
```
index.references.BioEntity = synonyms organism crossReferences 
```
So I commented it out on most mines built since that discovery (prod BeanMine being before it). So not a release version thing but an individual mine thing. (After convincing myself by looking at the diffs between tags that there was definitely nothing in the common code that did it).

And, yes, doing that work for EVERY BioEntity (including any SequenceFeature plus Proteins) is a bit exorbitant. BUT it turns out that that instruction is actually what creates the facet index in Solr. So what I'm going to do is turn it on for things of interest: Gene, Protein, etc. but leave it off of CDSRegion, CDS, GeneFlankingRegion, IntergenicRegion, and Exon. And anything else I don't explicitly list.
